### ISSUE TITLE
How do you use display updates with the JVM Repr?

### ISSUE THREAD

**Issue by rgbkrk** [jupyter/jvm-repr#9]
I'd like to be able to use [update display](http://jupyter-client.readthedocs.io/en/stable/messaging.html#update-display-data) in a Scala kernel. It involves using two things:

* `display_data` message with a `transient` field (alongside `data`, `metadata`) that has a `display_id`
* `update_display_data` message that also sets the `display_id` inside of `transient`

From a Python perspective, this is some of the interface provided to users:

![screen shot 2017-10-30 at 1 14 53 pm](https://user-images.githubusercontent.com/836375/32193388-54f27da2-bd74-11e7-858b-a58bf9078fbc.png)

**Comment by rdblue** [comment]
JVM repr is (currently) unrelated to display updates. That's like asking how to use `_repr_html_` to call `update_display` in python.

This project seems like a reasonable place for an API to handle display updates between Java libraries and Java kernels, but we'd have to think about how we want it to work first. Any ideas?

**Comment by rgbkrk** [comment]
Yeah, that makes sense. Since I don't know what would be idiomatic Java code for this, I'm unsure where to start.

My primary motivation is wanting to do HTML based progress bars for Spark (in Scala), which perhaps is best done within the kernel.

**Comment by rdblue** [comment]
I don't think we want to have kernels implement this directly, for the same reason that we want this lib for getting Java object representations. It would be better to have an API to broker the interaction between libraries and kernels.

Is this basically a way to add an ID and replace the HTML representation that is displayed?

**Comment by rgbkrk** [comment]
One of the terms for this is "named displays". You can change a particular `display_data` output at any point in time across an entire notebook, even persisting it to the document itself. The primary motivation for this protocol/message is one-way updates (without having to have widgets implemented in every kernel). This will end up being really great for dashboard usage.
